src/ripple/rpc/handlers/Tx.cpp

```diff
@@ -195,15 +204,25 @@ doTxHelp(RPC::Context& context, TxArgs const& args)
         std::pair<std::shared_ptr<Transaction>, std::shared_ptr<TxMeta>>;
 
     result.searchedAll = TxSearched::unknown;
-
     std::variant<TxPair, TxSearched> v;
+
+    if (args.ctid)
+    {
+        args.hash = context.app.getLedgerMaster().txnIDfromIndex(
+            args.ctid->first, args.ctid->second);
+
+        if (args.hash)
+            range =
+                ClosedInterval<uint32_t>(args.ctid->first, args.ctid->second);
+    }
+
     if (args.ledgerRange)
     {
-        v = context.app.getMasterTransaction().fetch(args.hash, range, ec);
+        v = context.app.getMasterTransaction().fetch(*(args.hash), range, ec);
```

Comment:
So, there's no need for parentheses here. `*args.hash` is sufficient. If you don't like that, consider `args.hash.value()`.

However I am concerned that this code may result in `args.hash` being accessed while unseated. I haven't dug deep down the codepath, but look at line 214 above. It suggests that it's possible for `args.hash` to be unseated, and there's no intervening assignment between that `if` and this code.

Sorry @RichardAH, I'm going to veto this until you can confirm that this is safe.

Reply:
Hi Nik

Thanks for the review.

Where it's called from (doTxJson) it's guaranteed to have populated either args.ctid or args.hash, but there's no harm in adding an additional check here.
https://github.com/XRPLF/rippled/blob/802144b48b3d7cb9ad2b037e7d2fff1450acbfd6/src/ripple/rpc/handlers/Tx.cpp#L351-L376

Reply:
Right, but _this_ code can be called with `hash` unseated, in favor of `ctid` (as the check on line 214 makes clear) so the extra check prior to the deference is needed. Thanks for adding it!

:+1: from me.

Reply:
Was a check going to be added?

This function, `doTxHelp`, is called with either `args.hash` or `args.ctid` set. If `args.ctid` is set, then the branch on 211 will be entered, assigning `args.hash` to the value returned by `txnIDfromIndex`. As @dangell7 [confesses](https://github.com/XRPLF/rippled/pull/4418#issuecomment-1610053304), and the code [shows](https://github.com/XRPLF/rippled/blob/1f61853422c6b185d901000d62c6b1d007cbeb6a/src/ripple/app/ledger/impl/LedgerMaster.cpp#L2378), that function can `return {}`, which is an empty `std::optional`. That case will lead to UB on line 222 (in the new code) when it is dereferenced.